lib/modem_info/modem_info.c

```diff
@@ -720,6 +733,144 @@ int modem_info_rsrp_register(rsrp_cb_t cb)
 	return 0;
 }
 
+/*
+ * Obtain and copy out a string value from a response to an AT command.
+ * One buffer is used internally and dimensioned for the known maximum size of
+ * the value, while the other is the user provided target buffer.
+ *
+ * cmd     Command to execute
+ * fmt     Format string by which to extract 1 string from response
+ * tmp_buf Buffer to hold response from nrf_modem_scanf
```

Comment:
Why do we need `tmp_buf`? Couldn't we just use `out_buf` directly, or am I missing something? Is it to handle a potential buffer overflow? You're already specifying the width of the string arguments where this function is called, so shouldn't it be enough to check the buffer length against the necessary buffer length? Either in the function calling `at_scanf_string` or by passing expected response length as an argument.